src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolClearerPass.php

```diff
@@ -31,9 +31,9 @@ public function process(ContainerBuilder $container)
         foreach ($container->findTaggedServiceIds('cache.pool.clearer') as $id => $attr) {
             $clearer = $container->getDefinition($id);
             $pools = array();
-            foreach ($clearer->getArgument(0) as $id => $ref) {
-                if ($container->hasDefinition($id)) {
-                    $pools[$id] = new Reference($id);
+            foreach ($clearer->getArgument(0) as $name => $ref) {
+                if ($container->hasDefinition($id = (string) $ref)) {
+                    $pools[$name] = new Reference($id);
```

Comment:
I found this change a bit hard to read as `$id` is used in both loops. Can we solve that by renaming one of them?

Reply:
updated